codegen/src/main/resources/software/amazon/awssdk/codegen/rules/AuthSchemeUtils.java.resource

```diff
@@ -50,56 +55,71 @@ public final class AuthSchemeUtils {
+                    for (int j = 0; j < signingRegionSetArray.size(); ++j) {
+                        schemeBuilder.addSigningRegion(signingRegionSetArray.get(j).expectString());
                     }
+                }
 
-                    authSchemes.add(schemeBuilder.build());
+                Value disableDoubleEncoding = scheme.get(Identifier.of("disableDoubleEncoding"));
+                if (disableDoubleEncoding != null) {
+                    schemeBuilder.disableDoubleEncoding(disableDoubleEncoding.expectBool());
                 }
-                break;
-                case SIGV4_NAME: {
-                    SigV4AuthScheme.Builder schemeBuilder = SigV4AuthScheme.builder();
 
-                    Value signingName = scheme.get(Identifier.of("signingName"));
-                    if (signingName != null) {
-                        schemeBuilder.signingName(signingName.expectString());
-                    }
+                authSchemes.add(schemeBuilder.build());
+            }
+                break;
+            case SIGV4_NAME: {
+                SigV4AuthScheme.Builder schemeBuilder = SigV4AuthScheme.builder();
 
-                    Value signingRegion = scheme.get(Identifier.of("signingRegion"));
-                    if (signingRegion != null) {
-                        schemeBuilder.signingRegion(signingRegion.expectString());
-                    }
+                Value signingName = scheme.get(Identifier.of("signingName"));
+                if (signingName != null) {
+                    schemeBuilder.signingName(signingName.expectString());
+                }
 
-                    Value disableDoubleEncoding = scheme.get(Identifier.of("disableDoubleEncoding"));
-                    if (disableDoubleEncoding != null) {
-                        schemeBuilder.disableDoubleEncoding(disableDoubleEncoding.expectBool());
-                    }
+                Value signingRegion = scheme.get(Identifier.of("signingRegion"));
+                if (signingRegion != null) {
+                    schemeBuilder.signingRegion(signingRegion.expectString());
+                }
 
-                    authSchemes.add(schemeBuilder.build());
+                Value disableDoubleEncoding = scheme.get(Identifier.of("disableDoubleEncoding"));
+                if (disableDoubleEncoding != null) {
+                    schemeBuilder.disableDoubleEncoding(disableDoubleEncoding.expectBool());
                 }
+
+                authSchemes.add(schemeBuilder.build());
+            }
+                break;
+            default:
+                LOG.debug(() -> "Ignoring unknown auth scheme: " + authSchemeName);
                 break;
-                default:
-                    LOG.debug(() -> "Ignoring unknown auth scheme: " + authSchemeName);
-                    break;
             }
         }
         return authSchemes;
     }
+
+    public static <T extends Identity> void putSelectedAuthScheme(ExecutionAttributes attributes,
```

Comment:
This method is unrelated to endpoint rules, but I guess it is convenient to put this method in endpoint rules related utils class; instead of a new one or generate it in AuthSchemeInterceptorSpec itself.